internal/repository/validator.go

```diff
@@ -7,198 +7,339 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"time"
+	"log/slog"
 
 	. "github.com/cartesi/rollups-node/internal/node/model"
 	"github.com/jackc/pgx/v5"
 )
 
-const DefaultServiceTimeout = 5 * time.Minute
+var NilEpoch = errors.New("epoch cannot be nil")
 
-func (pg *Database) GetLastProcessedBlock(
+// GetOutputs returns outputs produced by inputs sent to the application
+// between start and end blocks, inclusive. Outputs are returned in ascending
+// order by index.
+func (pg *Database) GetOutputs(
```

Comment:
I think the name `GetOutputs` is too generic, it should be `GetOutputsBetweenBlocks` or something similar

Reply:
There's a discussion to be had about standardizing all these getter methods, but let's leave it for later. 👌🏽 I'll use `GetOutputsProducedInBlockRange`. Done in 16ea3f0a22d2cf364f83694b975ca45e00f68e24.